tests/unit/test_main.py

```diff
@@ -1,26 +1,28 @@
 """Test Main methods."""
 from unittest import TestCase
-from unittest.mock import MagicMock, create_autospec, patch
+from unittest.mock import MagicMock, create_autospec, patch, PropertyMock
 
 from pyof.foundation.network_types import Ethernet
 from pyof.v0x01.controller2switch.common import StatsType
 from pyof.v0x04.controller2switch.common import MultipartType
 
 from kytos.core.connection import ConnectionState
-from tests.helpers import (get_connection_mock, get_controller_mock,
-                           get_kytos_event_mock, get_switch_mock)
+from kytos.lib.helpers import (get_switch_mock, get_kytos_event_mock,
+                               get_connection_mock)
+from napps.kytos.of_core.utils import NegotiationException
+from tests.helpers import get_controller_mock
 
 
-# pylint: disable=protected-access
+# pylint: disable=protected-access, too-many-public-methods
 class TestMain(TestCase):
-    """docstring for TestMain."""
+    """Test the Main class."""
 
     def setUp(self):
         """Execute steps before each tests.
         Set the server_name_url from kytos/of_core
         """
-        self.switch_v0x01 = get_switch_mock("00:00:00:00:00:00:00:01")
-        self.switch_v0x04 = get_switch_mock("00:00:00:00:00:00:00:02")
+        self.switch_v0x01 = get_switch_mock("00:00:00:00:00:00:00:01", "v0x01")
+        self.switch_v0x04 = get_switch_mock("00:00:00:00:00:00:00:02", "v0x04")
```

Comment:
In these lines _of_version_ shouldn't be string, you can pass it directly:

```suggestion
        self.switch_v0x01 = get_switch_mock("00:00:00:00:00:00:00:01", 0x01)
        self.switch_v0x04 = get_switch_mock("00:00:00:00:00:00:00:02", 0x04)
```